pkg/registry_checker/checker.go

```diff
@@ -193,7 +194,17 @@ func NewRegistryChecker(
 	}
 	rc.controllerIndexers.cronJobIndexer = rc.cronJobsInformer.Informer().GetIndexer()
 
-	rc.controllerIndexers.secretIndexer = rc.secretsInformer.Informer().GetIndexer()
+	namespace := "default"
+	// Create a context
+	ctx := context.TODO()
+	// Attempt to list secrets in the default namespace
+	_, enumerr := kubeClient.CoreV1().Secrets(namespace).List(ctx, metav1.ListOptions{})
```

Comment:
It seems like requesting all secrets on startup is an expensive operation and can be tricky. If pod is restarted multiple times, it will increase the control plane pressure.

One way to resolve the problem is to set the resource version in list options to 0. It will tell kube-apiserver to serve the request from the cache. However, we will still receive all the secrets with the full body. Using [metadata client](https://pkg.go.dev/k8s.io/client-go/metadata) may help here.

Another solution is to use the subject access review instead of the list request. https://pkg.go.dev/k8s.io/client-go/kubernetes/typed/authorization/v1